packages/core-allocation/src/engine.ts

```
@@ -0,0 +1,231 @@
+                });
+                continue; // or degrade
+            }
+
+            let costEntries: AllocationLedgerEntry[] = [];
+
+            switch (policy.type) {
+                case 'PRO_RATA':
+                    costEntries = this.allocateProRata(agreement, cost, period);
+                    break;
+                case 'BY_USAGE':
+                    const usageResult = this.allocateByUsage(agreement, cost, usageEvents, period);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)